tests/data/test_temporal_random_walker.py

```diff
@@ -17,19 +17,33 @@
 import pandas as pd
 import pytest
 import numpy as np
-import networkx as nx
 from stellargraph.data.explorer import TemporalRandomWalk
 from stellargraph.core.graph import StellarGraph
 
 
+def create_temporal_graph(nodes, edges):
+    edge_cols = ["source", "target", "weight"]
+    return StellarGraph(
+        nodes=pd.DataFrame(index=nodes), edges=pd.DataFrame(edges, columns=edge_cols),
+    )
+
+
 @pytest.fixture()
 def temporal_graph():
     nodes = [1, 2, 3, 4, 5, 6]
     edges = [(1, 2, 5), (2, 3, 2), (2, 4, 10), (4, 5, 3), (4, 6, 12)]
-    edge_cols = ["source", "target", "weight"]
-    return StellarGraph(
-        nodes=pd.DataFrame(index=nodes), edges=pd.DataFrame(edges, columns=edge_cols),
+    return create_temporal_graph(nodes, edges)
+
+
+def temporal_graph_negative_times(num_neg_edges):
+    nodes = [1, 2, 3, 4, 5, 6]
+    edges = np.hstack(
+        [
+            np.random.choice(nodes, size=(num_neg_edges * 2, 2)),
+            np.repeat([[-1], [1]], num_neg_edges, axis=0),
```

Comment:
Totally minor and potentially not an improvement, but this could be:

```suggestion
            np.repeat([-1, 1], num_neg_edges)[:, np.newaxis],
```

if you were so inclined.

Reply:
I like that it avoids defining the nested lists so I'll use your suggestion